File: app/src/main/java/com/lekai/root/iaddcontacts/Adapters/ContactAdapter.java
package com.lekai.root.iaddcontacts.Adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import com.lekai.root.iaddcontacts.ContactModel;
import com.lekai.root.iaddcontacts.R;

/**
 * Created by root on 4/21/17.
 */

public class ContactAdapter extends RecyclerView.Adapter<ContactAdapter.ContactViewHolder>{
    private Context mContext;
    int mCount ;
    ContactModel[] contacts;
    String name;
    String phone;
    public ContactAdapter(Context context, int count){
        mContext = context;
        mCount = count;
    }
    @Override
    public ContactViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(mContext);
        View view = inflater.inflate(R.layout.each_contact_view,parent,false);
        return new ContactViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ContactAdapter.ContactViewHolder holder, final int position) {
        contacts = new ContactModel[10];
        holder.contactPhone.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                name = holder.getName();
                phone = holder.getPhone();
                if(name.length()!=0 && phone.length() !=0) {
                    ContactModel contactModel = new ContactModel(name, phone);
                    contacts[position] = contactModel;
                }
            }

            @Override
            public void afterTextChanged(Editable s) {}
        });
    }
    @Override
    public int getItemCount() {
        return mCount;
    }
    public class ContactViewHolder extends RecyclerView.ViewHolder {
        EditText contactName;
        EditText contactPhone;
        public ContactViewHolder(View itemView) {
            super(itemView);
            contactName = (EditText) itemView.findViewById(R.id.contact_name);
            contactPhone = (EditText) itemView.findViewById(R.id.contact_phone);
        }
        public String getName(){
            return contactName.getText().toString();
        }
        public String getPhone(){
            return contactPhone.getText().toString();
        }
    }

    public ContactModel[] getAllContacts(){
        return contacts;
    }
}

File: app/src/main/java/com/lekai/root/iaddcontacts/ui/adapter/ContactAdapter.kt
package com.lekai.root.iaddcontacts.ui.adapter

import android.content.Context
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import androidx.recyclerview.widget.RecyclerView
import com.lekai.root.iaddcontacts.R
import com.lekai.root.iaddcontacts.models.ContactModel
import kotlinx.android.synthetic.main.each_contact_view.view.*

class ContactAdapter(private val context: Context, private val mCount : Int) :
        RecyclerView.Adapter<ContactAdapter.ContactViewHolder>() {
    var allContacts : Array<ContactModel?>? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactViewHolder  =
            ContactViewHolder(LayoutInflater.from(context).inflate(R.layout.each_contact_view, parent, false))

    override fun getItemCount(): Int  = mCount

    override fun onBindViewHolder(holder: ContactViewHolder, position: Int) {
        allContacts = arrayOfNulls(10)
        holder.phoneEditText?.addTextChangedListener(object : TextWatcher {
            override fun afterTextChanged(s: Editable?) {}

            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}

            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                val name =  holder.contactNameEditText!!.text.toString()
                val phone = holder.phoneEditText!!.text.toString()
                if (name.isNotEmpty() && phone.isNotEmpty()){
                    val contactModel = ContactModel(name, phone)
                    allContacts!![position] = contactModel
                }}

        })
    }

    inner class ContactViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
         var contactNameEditText: EditText? = itemView.contact_name
         var phoneEditText: EditText? = itemView.contact_phone
    }
}